Entities/AtomGroup.h

```diff
@@ -1,662 +1,387 @@
+		/// </summary>
+		/// <returns>The ratio of Atoms on top of terrain pixels, form 0 to 1.0.</returns>
+		float RatioInTerrain() const;
+
+		/// <summary>
+		/// Checks whether any of the Atoms in this AtomGroup are on top of terrain pixels, and if so, attempt to move the OwnerMO out so none of the Atoms are inside any terrain pixels anymore.
+		/// </summary>
+		/// <param name="position">Current position of the owner MO.</param>
+		/// <param name="strongerThan">Only attempt to move out of materials stronger than this specific ID.</param>
+		/// <returns>Whether any intersection was successfully resolved. Will return true even if there wasn't any intersections to begin with.</returns>
+		bool ResolveTerrainIntersection(Vector &position, unsigned char strongerThan = 0) const;
+
+		/// <summary>
+		/// Checks whether any of the Atoms in this AtomGroup are on top of MOSprites, and if so, attempt to move the OwnerMO out so none of the Atoms are inside the other MOSprite's silhouette anymore.
+		/// </summary>
+		/// <param name="position">Current position of the owner MO.</param>
+		/// <param name="rotation">Current rotation of the owner MO.</param>
+		/// <returns>Whether all intersections were successfully resolved.</returns>
+		bool ResolveMOSIntersection(Vector &position);
+#pragma endregion
+
+#pragma region Debug
+		/// <summary>
+		/// Draws this AtomGroup's current graphical debug representation to a BITMAP of choice.
+		/// </summary>
+		/// <param name="targetBitmap">A pointer to a BITMAP to draw on.</param>
+		/// <param name="targetPos">The absolute position of the target bitmap's upper left corner in the Scene.</param>
+		/// <param name="useLimbPos">Whether to use the limb position of this AtomGroup, or the owner's position.</param>
+		/// <param name="color">The color to draw the Atoms' pixels as.</param>
+		void Draw(BITMAP *targetBitmap, const Vector &targetPos = Vector(), bool useLimbPos = false, unsigned char color = 34) const;
+#pragma endregion
+
+	protected:
+
+		static Entity::ClassInfo m_sClass; //!< ClassInfo for this class.
+
+		std::list<Atom *> m_Atoms; //!< List of Atoms that constitute the group. Owned by this.
+		std::unordered_map<long, std::list<Atom *>> m_SubGroups; //!< Sub groupings of Atoms. Points to atoms owned in m_Atoms. Not owned.
+
+		MOSRotating *m_OwnerMO; //!< The owner of this AtomGroup. The owner is obviously not owned by this AtomGroup.
+		const Material *m_Material; //!< Material of this AtomGroup
+
+		bool m_AutoGenerate; //!< // Whether or not the Atoms were automatically generated based on a sprite, or manually defined.
+		int m_Resolution; //!< Resolution, or density of the Atoms in representing a MOSRotating's silhouette. 0 means the Atom config wasn't generated by a silhouette, but defined manually.
+		int m_Depth; //!< Depth, or how deep into the bitmap of the owning MO's graphical representation the Atoms of this AtomGroup are located, in pixels.
```

Comment:
This comment is okay as-is, but it'd be nice if it explained that depth means moving the atoms into the object. I'm not sure how you should rewrite it to say this though, so if you can't think of anything, leave it be.